src/Components/Shared/src/WebEventData.cs

```diff
@@ -245,4 +258,22 @@ private static ChangeEventArgs DeserializeChangeEventArgs(string eventArgsJson)
             return changeArgs;
         }
     }
+
+    [JsonSerializable(typeof(WebEventDescriptor), GenerationMode = JsonSourceGenerationMode.MetadataAndSerialization)]
```

Comment:
Is there a reason you aren't just letting it default the "GenerationMode"?

```suggestion
    [JsonSerializable(typeof(WebEventDescriptor)]
```